src/control/destination/background/mod.rs

```diff
@@ -208,16 +233,30 @@ where
                                 },
                                 Exists::No | Exists::Unknown => (),
                             }
-                            set.responders.push(resolve.responder);
+
+                            // If the destinationSet was previously unable to
+                            // make a query due to insufficient capacity, try
+                            // to open a new query if possible.
+                            if occ.get().wants_query() {
```

Comment:
for instance, i think we could basically remove this comment by renaming the fn to be more descriptive